taichi/ir/snode.cpp

```diff
@@ -54,30 +54,26 @@ SNode &SNode::create_node(std::vector<Axis> axes,
       throw TaichiRuntimeError(
           "Every dimension of a Taichi field should be positive");
     }
-    auto &ind = axes[i];
-    new_node.extractors[ind.value].activate(
+    int ind = axes[i].value;
+    auto end = new_node.physical_index_position + new_node.num_active_indices;
+    bool is_first_division =
+        std::find(new_node.physical_index_position, end, ind) == end;
+    if (is_first_division) {
+      new_node.physical_index_position[new_node.num_active_indices++] = ind;
+    } else {
+      if (!bit::is_power_of_two(sizes[i])) {
```

Comment:
Shall we limit this check to packed mode only? In non-packed mode, we always pad `sizes` to the power of 2, so this limitation should've never applied?

Reply:
You're right. Although non-packed mode will be removed, this PR should make a proper intermediate state.